Code/Framework/AzCore/AzCore/Memory/IAllocator.h

```diff
@@ -73,8 +131,10 @@ namespace AZ
         IAllocator& operator=(IAllocator&&) = delete;
         virtual ~IAllocator() = default;
 
-        virtual pointer allocate(size_type byteSize, align_type alignment = 1) = 0;
-        virtual void deallocate(pointer ptr, size_type byteSize = 0, align_type alignment = 0) = 0;
+        //! Allocates memory and returns the address + the amount of memory allocated
+        virtual AllocateAddress allocate(size_type byteSize, align_type alignment = 1) = 0;
+        //! Deallocates memory at the ptr address and returns the amount of memory that was allocated
+        virtual size_type deallocate(pointer ptr, size_type byteSize = 0, align_type alignment = 0) = 0;
```

Comment:
I think we should consider incrementing the framework minor version and the engine's too in engine.json so devs have a version to point to at which `AllocateAddress` exists.

Reply:
Yeah I can do that